meos/src/point/tpoint_spatialrels.c

```diff
@@ -367,116 +377,31 @@ acontains_geo_tpoint(const GSERIALIZED *gs, const Temporal *temp)
-      return ret_loop;
-  }
-  return ! ret_loop;
-}
-
-/**
- * @brief Return true if a temporal sequence set point and a geometry are ever
- * disjoint
- * @param[in] ss Temporal point
- * @param[in] geo Geometry
- * @param[in] func PostGIS function to be used for instantaneous sequences
- * @param[in] ever True for the ever semantics, false for the always semantics
- */
-bool
-ea_disjoint_tpointseqset_geo(const TSequenceSet *ss, Datum geo,
-  datum_func2 func, bool ever)
-{
-  bool ret_loop = ever ? true : false;
-  for (int i = 0; i < ss->count; i++)
-  {
-    bool res = ea_disjoint_tpointseq_geo(TSEQUENCESET_SEQ_N(ss, i), geo, func,
-      ever);
-    if ((ever && res) || (! ever && ! res))
-      return ret_loop;
-  }
-  return ! ret_loop;
-}
-
-/**
- * @ingroup meos_internal_temporal_spatial_rel_ever
- * @brief Return 1 if a temporal point and a geometry are ever disjoint,
- * 0 if not, and -1 on error or if the geometry is empty
- * @param[in] temp Temporal point
- * @param[in] gs Geometry
- * @param[in] ever True for the ever semantics, false for the always semantics
- * @csqlfn #Edisjoint_tpoint_geo()
- */
-Datum
-ea_disjoint_tpoint_geo(const Temporal *temp, const GSERIALIZED *gs, bool ever)
-{
-  /* Ensure validity of the arguments */
-  if (! ensure_valid_tpoint_geo(temp, gs) || gserialized_is_empty(gs))
-    return Int32GetDatum(-1);
-
-  datum_func2 func = get_disjoint_fn_gs(temp->flags, gs->gflags);
-  bool result;
-  assert(temptype_subtype(temp->subtype));
-  switch (temp->subtype)
-  {
-    case TINSTANT:
-      result = ea_disjoint_tpointinst_geo((TInstant *) temp, PointerGetDatum(gs),
-        func);
-      break;
-    case TSEQUENCE:
-      result = ea_disjoint_tpointseq_geo((TSequence *) temp, PointerGetDatum(gs),
-        func, ever);
-      break;
-    default: /* TSEQUENCESET */
-      result = ea_disjoint_tpointseqset_geo((TSequenceSet *) temp,
-        PointerGetDatum(gs), func, ever);
-  }
-  return result ? Int32GetDatum(1) : Int32GetDatum(0);
-}
-
 /**
  * @ingroup meos_temporal_spatial_rel_ever
  * @brief Return 1 if a temporal point and a geometry are ever disjoint,
  * 0 if not, and -1 on error or if the geometry is empty
  * @param[in] temp Temporal point
  * @param[in] gs Geometry
+ * @note eDisjoint(tpoint, geo) is equivalent to NOT covers(geo, traj(tpoint))
+ * @note The function does not accept geography since it is based on the
+ * PostGIS ST_Covers function provided by GEOS
  * @csqlfn #Edisjoint_tpoint_geo()
  */
 int
 edisjoint_tpoint_geo(const Temporal *temp, const GSERIALIZED *gs)
 {
-  return DatumGetInt32(ea_disjoint_tpoint_geo(temp, gs, EVER));
+  /* Ensure validity of the arguments */
+  if (! ensure_not_geodetic(temp->flags) || gserialized_is_empty(gs) ||
+      ! ensure_valid_tpoint_geo(temp, gs))
+    return -1;
+  datum_func2 func = &geom_covers;
+  int result = spatialrel_tpoint_traj_geo(temp, gs, (Datum) NULL,
+    (varfunc) func, 2, INVERT);
+  return result ? 0 : 1;
```

Comment:
This will return 1 if result = -1, which is not wanted.
We probably want `return result < 0 ? result : (result > 0 ? 0 : 1);`
Should we make macros for this?

Reply:
Something like:
```C
#define INVERT_RESULT(result) (result < 0 ? -1 : (result > 0) ? 0 : 1)
```

Could be another name, this is just the first idea I had.
We just need to make sure we only call it as `INVERT_RESULT(var)` and never as `INVERT_RESULT(f(param))`, otherwise f could be evaluated twice.

I'm fine with avoiding the macro and keeping the expressions duplicated, but we need to make sure they're all the same.